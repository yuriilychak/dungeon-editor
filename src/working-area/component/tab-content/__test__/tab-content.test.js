import React from "react";

import Close from "@material-ui/icons/Close";
import {createMount} from "@material-ui/core/test-utils";

import TabContent from "../tab-content";

describe("tab-content test",()=> {
    const onClose = jest.fn();
    const wrapper = createMount()(
        <TabContent
            title="test"
            icon="test"
            onClose={onClose}
        />
    );

    it ( "tab-content snapshot", () => expect(wrapper.html()).toMatchSnapshot());

    it ( "tab-content dispatch close", () => {
        const closeButton = wrapper.find(Close);

        closeButton.simulate("click");
        expect(onClose).toHaveBeenCalled();
    });
});
