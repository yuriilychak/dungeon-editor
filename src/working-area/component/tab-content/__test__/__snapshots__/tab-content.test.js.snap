// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab-content test tab-content snapshot 1`] = `"<div class=\\"tab-content-root\\"><img src=\\"undefined/static/icon/test.svg\\" style=\\"width: 10px; height: 10px;\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\"><span class=\\"tab-content-title\\">test</span><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></div>"`;
